src/OpenDiffix.Core/Aggregator.fs

```diff
@@ -246,8 +246,8 @@ type private DiffixCountDistinctNoise(aidsCount) =
       let anonContext = unwrapAnonContext anonContext
 
       match Anonymizer.countDistinct aggContext.AnonymizationParams anonContext this.AidsCount this.State with
-      | Anonymizer.AnonymizedResult.NotEnoughAIDVs -> Null
-      | Anonymizer.AnonymizedResult.Ok { NoiseSD = noiseSD } -> Real noiseSD
+      | None -> Null
```

Comment:
IIRC this has been given a name on purpose, to have the reasoning behind the `Null` case spelled out. Otherwise it's just too many `None` and `Null` cases, making the code hard to follow

Reply:
I didn't find it helpful at all. It acts like a renamed `Option` type, but with a lot more verbosity.
It provides a single error case, for which we don't do anything special, except returning `Null`.
I think it hinders more than it helps.

Reply:
> It acts like a renamed `Option` type, but with a lot more verbosity

That's exactly what it is intended to be and I found it very helpful, in contrast to the obscure `Null`s and `None`s being passed around here. It spared me a lot of head-scratching and look-ups.

Note that our course of action in these `match` clauses is very specific to our logic, so a specific value is better than a generic (`None`). Same goes for the conditions in `Anonymizer` which emit these `NotEnoughAIDVs`.

Reply:
I don't understand what we get by explicitly exposing the reason for not being able to produce an anonymized result to the callers. It is irrelevant to them, since there is only one possible cause and they also don't take any specific action on it. I find it adds noise to the codebase for little to no gain.